sound/soc/sof/ipc4-topology.h

```diff
@@ -163,15 +163,16 @@ struct sof_ipc4_pin_format {
  * struct sof_ipc4_available_audio_format - Available audio formats
  * @out_audio_fmt: Available output audio format
  * @sink_audio_fmts: Available sink(input) audio formats
- * @ref_audio_fmt: Reference audio format to match runtime audio format
+ * @format_list_to_search: List of audio formats to match runtime params against. This points to
+ * either @out_audio_fmt or @input_audio_fmts
```

Comment:
Something is not quite right with this definition.

in `sof_ipc4_init_audio_fmt()` we are checking the `format_list_to_search` for a matching format, that is reasonable.
When we found it we will do this:
```c
memcpy(&base_config->audio_fmt, &available_fmt->sink_audio_fmts[i].audio_fmt, sizeof(*out_format));
```
But based on the naming of `out_audio_fmt` and `input_audio_fmts`, the `input_audio_fmts` holds a single format, while the input side have an array?
How this is safe if the `format_list_to_search` pointing to `out_audio_fmt`?

Reply:
How it was safe to begin with?